atlas-spec/mpc-engine/src/party.rs

```diff
@@ -99,24 +97,95 @@ impl Party {
             if i == self.id {
                 continue;
             }
-
-            self.channels.parties[i]
-                .send(round_function(self.id, i))
-                .unwrap();
+            let left_input = [1u8, 1u8, 1u8];
+            let right_input = [9u8, 9u8, 9u8];
+            self.ot_send(i, &left_input, &right_input)?;
         }
 
         // Wait for the messages sent by later parties.
         for _i in self.id + 1..num_parties {
-            let msg = self.channels.listen.recv().unwrap();
-            self.process_message(msg);
+            let choose_left = self.entropy.bit()?;
+            ot_results.push(self.ot_receive(choose_left)?);
         }
+
+        Ok(ot_results)
     }
 
-    /// Process an incoming message.
-    fn process_message(&mut self, msg: MPCMessage) {
-        match msg {
-            _ => todo!(),
-        }
+    /// Initiate an OT session as the Sender.
+    ///
+    /// The sender needs to provide two inputs to the OT protocol and receives
+    /// no output.
+    fn ot_send(&mut self, i: usize, left_input: &[u8], right_input: &[u8]) -> Result<(), Error> {
+        let (own_sender, own_receiver) = mpsc::channel::<SubMessage>();
+        let (their_sender, their_receiver) = mpsc::channel::<SubMessage>();
+
+        let channel_msg = Message {
+            from: self.id,
+            to: i,
+            payload: MessagePayload::SubChannel(own_sender, their_receiver),
+        };
+        self.channels.parties[i].send(channel_msg).unwrap();
+
+        let dst = format!("OT-{}-{}", self.id, i);
+        let (ot_sender, ot_commit) =
+            crate::primitives::ot::OTSender::init(&mut self.entropy, dst.as_bytes())?;
+        their_sender.send(SubMessage::OTCommit(ot_commit)).unwrap();
+        let receiver_msg = own_receiver.recv().unwrap();
+        match receiver_msg {
+            SubMessage::OTSelect(selection) => {
+                let send =
+                    ot_sender.send(left_input, right_input, &selection, &mut self.entropy)?;
+                their_sender.send(SubMessage::OTSend(send)).unwrap();
+            }
+            _ => panic!(
+                "Party {} - Unexpected receiver message in OT:\n{receiver_msg:#?}",
+                self.id
+            ),
+        };
+        Ok(())
+    }
+
+    /// Listen for an OT initiation as the receiver.
+    ///
+    /// The receiver needs to provide a choice of left or right output to the
+    /// protocol and receives the chosen sender input.
+    fn ot_receive(&mut self, choose_left: bool) -> Result<Vec<u8>, Error> {
+        let channel_msg = self.channels.listen.recv().unwrap();
+
+        Ok(match channel_msg {
```

Comment:
That's a pretty big `Ok()` ... This would look a little nicer when you moved that inside or stored the result of this `match`.

Reply:
I've addressed this with #71, by turning these `match`es into `if let`s since I'm only matching on one thing at a time anyway.